src/NuGet.Clients/NuGet.MSSigning.Extensions/NuGet.MSSigning.Extensions.csproj

```diff
@@ -41,4 +31,35 @@
   </ItemGroup>
 
   <Import Project="$(BuildCommonDirectory)common.targets" />
+
+  <Target Name="ILMergeNuGetMssignExe" AfterTargets="Build">
+    <ItemGroup>
+      <BuildArtifacts Include="$(OutputPath)\*.dll" Exclude="@(MergeExclude)" />
+    </ItemGroup>
+    <Error Text="Build dependencies are inconsistent with mergeinclude specified in ilmerge.props" Condition="'@(BuildArtifacts-&gt;Count())' != '@(MergeInclude-&gt;Count())'" />
+    <PropertyGroup>
+      <PathToBuiltNuGetExe>$(OutputPath)NuGet.exe</PathToBuiltNuGetExe>
```

Comment:
the built nuget.exe won't be at this path. it will be artifactsdirectory\vsixoutputdirname

Reply:
discussed with @zhili1208 , he is not using the ilmerged nuget.exe , instead merging all the assemblies from scratch.